subprojects/core/src/main/java/org/gradle/execution/plan/SingleFileTreeElementMatchUtil.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.execution.plan;
+
+import org.gradle.api.file.FileTreeElement;
+import org.gradle.api.file.RelativePath;
+import org.gradle.api.specs.Spec;
+import org.gradle.internal.UncheckedException;
+import org.gradle.internal.file.Stat;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+
+public class SingleFileTreeElementMatchUtil {
```

Comment:
Yeah, I'd really like a better name for this thing. The strange thing is here that you'd also pass the `Spec` as a first argument to the `match` method. Would it be better to convert the `Spec` into a `SingleFileTreeElementMatcher` and then call a method on this thing via `matcher`?

So something like:
```
Spec<FileTreeElement> spec;
SingleFileTreeElementMatcherFactory matcherFactory;
SingleFileTreeElementMatcher matcher = matcherFactory.createFor(spec)
matcher.matches(file, relativePath);
```

I suppose that is too complicated for this small use case, so maybe just going with `SingleFileTreeElementMatcher` is good. WDYT?

Reply:
It feels too much work for this small use case to me too. 

If we will add more "helper" methods to this class maybe `Util` is better, but if it will be just this method then I think `Matcher` is good.

Reply:
`SingleFileTreeElementMatcher` it is.